### ISSUE TITLE
Definition of lithostatic pressure

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#4508]
@cedrict and I were having a discussion about the lithostatic pressure. Usually, this is defined as
```
  p(z) = \int_z^top rho(z) g(z) dz
```
and the interpretation is that it is the weight of the overlying rock in a column over a point at vertical position `z`.

But this definition is not actually correct. There are many cases where there is no actual *static* pressure because material is not stably layered, but apart from that, there is also the issue that the definition above is only true for a Cartesian coordinate system where the gravity vector is always pointing in the same direction. On the other hand, in a *spherical* domain where the gravity points radially inward, the definition above should really read
```
  p(r) = \int_r^R rho(s) g(s)  s^{d-1}/r^{d-1} ds
```
to account for the fact that the "column of material" above a point at radius `r` geometrically widens the higher we go.

I don't think we account for this.

**Comment by bobmyhill** [comment]
> the interpretation is that it [the lithostatic pressure] is the weight of the overlying rock in a column over a point at vertical position z. But this definition is not actually correct. There are many cases where there is no actual static pressure...

I guess that depends on your definition of "definition" :D
I'd say that I usually *define* the lithostatic pressure as \int rho g dz, where rho could have contributions from rock, fluid or gas. That value may or may not accurately reflect the true total pressure, but it's a useful value for comparison. 

For the second part of your post, are you suggesting that the classic hydrostatic equation (<img src="https://render.githubusercontent.com/render/math?math={\nabla P = \rho \mathbf{g}}#gh-light-mode-only"><img src="https://render.githubusercontent.com/render/math?math={\color{white}\nabla P = \rho \mathbf{g}}#gh-dark-mode-only">) is incorrect?

**Comment by bangerth** [comment]
Yes, `grad p = rho*g` is an equation that only has a solution if `rho * g` is the gradient of a potential. An example is a vertical gravity vector with a layered density field.

But the real point I wanted to make is that `grad p = rho*g` is not equivalent to `d/dr p = rho g_r` in a spherical coordinate system, I believe (but may also be wrong).

**Comment by bobmyhill** [comment]
> Yes, grad p = rho*g is an equation that only has a solution if rho * g is the gradient of a potential. An example is a vertical gravity vector with a layered density field.

True. We discussed this via email, but to repeat here, the "lithostatic pressure" defined in textbooks implicitly assumes a vertical gravity vector with a layered density field. We ignore any deviation from this away from a line that passes through our point of interest.

> But the real point I wanted to make is that grad p = rho*g is not equivalent to d/dr p = rho g_r in a spherical coordinate system, I believe (but may also be wrong).

Maybe I'm misunderstanding you, but the solution to a problem shouldn't be dependent on coordinate system. The mathematics may get harder (or easier), but the solution should stay the same. A simple counterexample to your claim would be the expressions for the pressure and gravitational potential inside a sphere of uniform density, from which you can demonstrate that `grad p = rho*g`.

**Comment by bangerth** [comment]
When I wrote "spherical coordinate system", I implied a gravity vector that is also pointing radially inward.

Whether I'm right or wrong we can easily test. If you take sphere, a constant-magnitude inward gravity, and constant density, then p increases linearly from the outer boundary if you compute it via `d/dx p = rho g_r`. Is this the correct pressure in such a sphere, though?

**Comment by cedrict** [comment]
the gradient operator in spherical coordinates for spherically symmetrical systems is simply d/dr so then one must solve dp/dr = rho g_r.

**Comment by bobmyhill** [comment]
@bangerth if you can show us what you mean by force balance on a fluid element in a sphere, then I'd be willing to read the derivation. It would overthrow an awful lot of geo, planetary and stellar physics...

**Comment by bangerth** [comment]
Hm, I guess I stand corrected. I took a piece of paper and worked this out for myself: Take a sphere for constant density, solve the Poisson equation for the gravity potential (which yields `phi = -2*pi*rho(r^2-rR)`), compute its gradient to get the gravity vector (`\vec g = 2*pi*rho(R-2r) \hat r` where `\hat r` is the unit radial vector), then solve
```
  p = -\int_r^R rho*\vec g . \hat r
```
and obtain
```
  p = -2*pi*rho* (r^2 - rR)
```
which is indeed the same as the gravity potential above. 

In other words, my intuition that one has to take into account the geometric widening of the cone that sits above an area element if we have a radial gravity vector is not actually correct. If that had been the case, then we would have to compute
```
  p = -\int_r^R rho(r')*\vec g(r) . \hat r  (r'^2/r^2)
```
which would have yielded a different pressure. But it appears as if that is not necessary.

So I guess that means that we can close this conversation :-)
